### PR TITLE
✅ More `RootModel` tests

### PR DESCRIPTION
**Pull request by lig**
## Change Summary

Add tests illustrating `RootModel` issues

## Related issue number

None

## Checklist

* [X] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @samuelcolvin

### CODE REVIEW

**Comment by lig** [comment]
please review

File: tests/test_root_model.py [hunk 1]
```diff
@@ -221,6 +232,145 @@ def double(cls, v):
...
+    assert m._private_attr_default == 'abc'
+    with pytest.raises(AttributeError, match='_private_attr'):
+        m._private_attr
+
+    m._private_attr = 7
+    m._private_attr_default = 8
+    m._other_private_attr = 9
+    # TODO: Should this be an `AttributeError`?
+    with pytest.raises(ValueError, match='other_attr'):
+        m.other_attr = 10
```
Comment (Kludex) [inline tests/test_root_model.py:256]: It shouldn't be `ValueError`, but not sure if it should be something different than `AttributeError`.

File: tests/test_root_model.py [hunk 2]
```diff
@@ -221,6 +232,145 @@ def double(cls, v):
...
+        _private_attr: str = PrivateAttr(default='abc')
+
+    m = Model(42)
+    assert m == Model(42)
+
+    m._private_attr = 'xyz'
+    assert m != Model(42)
+
+
+@pytest.mark.xfail(reason='TODO: raise error for `extra` with `RootModel`')
```
Comment (Kludex) [inline tests/test_root_model.py:313]: Why should it raise on this case as well?

File: tests/test_root_model.py [hunk 3]
```diff
@@ -221,6 +232,145 @@ def double(cls, v):
...
+
+
+def test_root_model_as_attr_with_validate_defaults():
+    class Model(BaseModel):
+        model_config = ConfigDict(validate_default=True)
+
+        rooted_value: RootModel[int] = 42
+
+    m = Model()
+    # assert m.rooted_value == RootModel[int](42)  # see `test_root_model_equality`
```
Comment (Kludex) [inline tests/test_root_model.py:339]: Should you add a TODO to not forget this, or have you already removed on the next PR?

**Comment by Kludex** [comment]
please update

**Comment by lig** [comment]
@Kludex please review

File: tests/test_root_model.py (hunk 3 above)
Comment (Kludex) [reply tests/test_root_model.py:339]: that was the second part of the question 👀

File: tests/test_root_model.py (hunk 1 above)
Comment (Kludex) [reply tests/test_root_model.py:256]: Are you going to use `AttributeError` on this PR then?

Comment (lig) [reply tests/test_root_model.py:256]: Not in this one. Maybe later. That is why I did put `TODO` here:)

**Pull request merged**